### ISSUE TITLE
Missing file in insert_editor_js breaks page editor

### ISSUE THREAD

**Issue by balazs-endresz** [wagtail/wagtail#4080]
### Issue Summary

If a non-existent file is referenced from an `insert_editor_js` hook then django won't render a large part of the page editor template at all. There's no indication of what caused this thanks to django suppressing exceptions from template includes. 

### Steps to Reproduce

1. reference a non-existent file in an `insert_editor_js` hook:
```
# wagtail_hooks.py
from django.contrib.staticfiles.templatetags.staticfiles import static
from django.utils.safestring import mark_safe
from wagtail.wagtailcore import hooks

@hooks.register('insert_editor_js')
def editor_js():
    return mark_safe('<script src="%s"></script>' % static('js/does-not-exist.js'))
```
2. set `DEBUG=False`
3. run collectstatic
4. try to edit a page: there will be lots of javascript errors on the console and e.g. image choosers won't work

### More details

When developing locally with `DEBUG=True` then `django.contrib.staticfiles.templatetags.staticfiles.static` won't raise an exception for missing files, instead it will return the path even if the file doesn't exist. This causes no other issues at all, it will be just a 404 later on.

However with `DEBUG=False` and using the manifest storage `django.contrib.staticfiles.templatetags.staticfiles.static` does raise an exception via `{% hook_output 'insert_editor_js' %}`: https://github.com/wagtail/wagtail/blob/master/wagtail/admin/templates/wagtailadmin/pages/_editor_js.html#L28
Which is currently ignored by django if `DEBUG=False` but at least this behaviour will change in django 2.1: https://docs.djangoproject.com/en/1.11/ref/templates/builtins/#include
Moreover, since `hook_output` is in a template include this exception will turn the whole `_editor_js.html` template into an empty string, not just the `hook_output` tag, omitting 10 more javascript files that are required for the page editor to work.

Maybe this doesn't affect many people and it really isn't Wagtail's fault but it isn't obvious at all why suddenly the page editor is broken when there are no exceptions raised with either `DEBUG=True/False`. I'm not sure if there's a good fix for this, or even if it's worth addressing it until django 2.1 but at least it might come in handy for others having the same issue.

### Technical details

* Django version: 1.11.5
* Wagtail version: 1.13.1

**Comment by lb-** [comment]
@balazs-endresz thanks for this issue and putting in some awesome detail.

Not sure on the solution here but wanted to add a note about what the Django docs state:
> Deprecated since version 1.11:
> Silencing exceptions raised while rendering the {% include %} template tag is deprecated. In Django 2.1, the exception will be raised.

Without some kind of exception to catch, I am not sure how we could check if the static files exist when the files are built up.

**Comment by balazs-endresz** [comment]
We could add a new include which contains only `{% hook_output 'insert_editor_js' %}`, so if there's a missing file then at least the rest of `_editor_js.html` wouldn't disappear. I can add a PR if you think it's worth addressing it.

**Comment by lb-** [comment]
@balazs-endresz

That sounds like an awesome solution, that way if the include fails it will only fail for content from the hooks. 

Would love a PR if you can, it will need to preserve any ordering that already happens. 

Can you also revise the  implementation of the other similar hooks. 

* insert_editor_css
* insert_editor_js
* insert_global_admin_css
* Insert_global_admin_js

Any other thoughts @gasman ?

**Comment by balazs-endresz** [comment]
I've added tests but also realised that an exception will still break all the other e.g. `insert_editor_js` hooks if only one of them raises an exception because they are all evaluated at the same place: https://github.com/wagtail/wagtail/blob/db6d36845f3f2c5d7009a22421c2efab9968aa24/wagtail/admin/templatetags/wagtailadmin_tags.py#L171

So, I'm not convinced a partial fix like this is all that helpful.

**Comment by lb-** [comment]
@balazs-endresz I think that will still be helpful though, it will break all hooks but at least that is easier to diagnose and won't break all css or js.

Also, with Django 2.0 - it will be easy to see where the error is coming from when errors are thrown, it will report that the hooks template is broken.

**Comment by gasman** [comment]
I think there's a strong case for saying: if it's going to break, better for it to break in a major conspicuous way, rather than causing subtle breakage on a few plugins that might sneak unnoticed onto production.

**Comment by gasman** [comment]
Yep, let's leave this to be fixed at the Django end.

**Comment by balazs-endresz** [comment]
I don't mind leaving it as it is either but it's going to sneak into production anyway thanks to the `serve()` function _not_ failing locally when `DEBUG = True` and the includes will be fixed only in Django 2.1 not 2.0. Not changing this at least means that it's easier to notice that something has gone wrong.
